When getting the org-agenda-list, in the start-day parameter is a function, call it.

This provides a way for the user to indicate what day the agenda should show.
It is useful to get a time log of yesterday for instance.
Index: devel/config/elfiles/org/lisp/org-agenda.el
===================================================================
--- devel.orig/config/elfiles/org/lisp/org-agenda.el	2013-01-07 18:04:19.528445996 +0100
+++ devel/config/elfiles/org/lisp/org-agenda.el	2013-01-07 18:04:35.816446511 +0100
@@ -3977,6 +3977,9 @@
 	      org-agenda-buffer-name))
     (org-agenda-prepare "Day/Week")
     (setq start-day (or start-day org-agenda-start-day))
+    (when (functionp start-day)
+      (setq start-day (funcall start-day))
+      )
     (if (stringp start-day)
 	;; Convert to an absolute day number
 	(setq start-day (time-to-days (org-read-date nil t start-day))))
